### ISSUE TITLE
string or query in combination with another string query criteria raises an exception

### ISSUE THREAD

**Issue by liorchen** [josiahcarlson/rom#57]
I have the following model:
```python
class ClientCommand(rom.Model):
    cmd_type = rom.Integer(required=True, index=True)
    app = rom.String(required=True, index=True)
    status = rom.String(required=True, default=COMMAND_STATUS.PENDING, index=True)
    client_id = rom.String(required=True, index=True)
    ts_queued = rom.DateTime(default=datetime.datetime.utcnow, index=True)
    last_sent = rom.DateTime()
    sent_count = rom.Integer(default=0)
    initiator_type = rom.Integer(default='server')
    error_code = rom.Integer()
    raw = rom.Text(required=True)
    initiator_id = rom.String()
    client_identifier = rom.String(required=False, index=True)
```

1. When I call ```ClientCommand.query.filter(client_id='548ef7ee7b77b93ab41ksjh3').execute()``` - it works
2. When I call ```ClientCommand.query.filter(status=['1','2']).execute()``` - it works
3. When I call ```ClientCommand.query.filter(status='1', client_id='548ef7ee7b77b93ab41ksjh3').execute()``` - it works.
4. When I call ```ClientCommand.query.filter(status=['1', '2'], client_id='548ef7ee7b77b93ab41ksjh3').execute()``` I get the following exception:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/rom/__init__.py", line 1143, in execute
    return self._model.get(self._search())
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/rom/__init__.py", line 1077, in _search
    _connect(self._model), self._filters, self._order_by, *limit)
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/rom/index.py", line 282, in search
    pipe, intersect, temp_id = self._prepare(conn, filters)
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/rom/index.py", line 184, in _prepare
    sizes = list(enumerate(pipe.execute()))
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/redis/client.py", line 2578, in execute
    return execute(conn, stack, raise_on_error)
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/redis/client.py", line 2492, in _execute_transaction
    self.raise_first_error(commands, response)
  File "/var/www/zconsole.vnv/lib/python2.7/site-packages/redis/client.py", line 2526, in raise_first_error
    raise r
redis.exceptions.ResponseError: Command # 1 (ZCARD ClientCommand:status:1:idx) of pipeline caused error: WRONGTYPE Operation against a key holding the wrong kind of value
```

Any idea what is the problem?
Any workaround?

**Comment by josiahcarlson** [comment]
This is the first time I've seen the issue, let me look into it and get back to you hopefully later today.

**Comment by josiahcarlson** [comment]
Fundamentally, this issue is the result of a bad API.

When passing arguments using `col=[v1, v2]`, the meaning of the argument can be one of either "find me a range of things between these two values" and "find me exactly these two values". This wouldn't be too bad, but "find me exactly two values" has a different underlying meaning depending on whether the index is a numeric index, unique index, prefix/suffix index, or full text index, or even in some cases when a column has more than one index (an individual column can have up to 4 indexes built on it to service different types of queries)!

For the short term, I'm going to fix the exception itself with a Lua script, but longer term I'm going to need to work on the API.

**Comment by josiahcarlson** [comment]
Fixed in 0.29.4, available now.

**Comment by liorchen** [comment]
awesome! thanks :+1:
